src/libraries/System.Private.CoreLib/src/System/Globalization/JapaneseCalendar.Icu.cs

```diff
@@ -47,16 +44,37 @@ public partial class JapaneseCalendar : Calendar
                     break;
                 }
 
-                eras.Add(new EraInfo(i, dt.Year, dt.Month, dt.Day, dt.Year - 1, 1, lastMaxYear - dt.Year + 1,
-                    eraNames![i], GetAbbreviatedEraName(eraNames, i), abbrevEnglishEraNames![i]));
+                eras.Add(new EraInfo(i, dt.Year, dt.Month, dt.Day, dt.Year - 1, 1, lastMaxYear - dt.Year + 1, eraNames![i], GetAbbreviatedEraName(eraNames, i), ""));
 
                 lastMaxYear = dt.Year;
             }
 
+            string[] abbrevEnglishEraNames;
+            if (!CalendarData.EnumCalendarInfo("ja", CalendarId.JAPAN, CalendarDataType.AbbrevEraNames, out abbrevEnglishEraNames!))
+            {
+                // Failed to get English names. fallback to hardcoded data.
+                abbrevEnglishEraNames = s_abbreviatedEnglishEraNames;
+            }
+
+            if (abbrevEnglishEraNames[abbrevEnglishEraNames.Length - 1].Length != 1 || abbrevEnglishEraNames[abbrevEnglishEraNames.Length - 1][0] > '\u007F')
```

Comment:
Just to understand this, why would having the Length != 1 of the abbrevEnglishEraNames on the last element or having a character in first position greater than `DELETE` character is an indicator that we couldn't get the abbreviated english names?

Reply:
The Japanese English abbreviated names so far is only one letter. In some versions of ICU versions (e.g. ver 50) it doesn't return these Latin letters. The check here is indicating if we have Latin or non Latin names. The check here is enough. I may remove the length check just in case in the future we'll get more than one letter (which unlikely but it is possible).

Reply:
I see, makes sense. Thanks for explaining!